model/test_results.go

```diff
@@ -295,42 +328,116 @@ func (t *TestResults) Download(ctx context.Context) ([]TestResult, error) {
 		t.ID = t.Info.ID()
 	}
 
-	var results testResultsDoc
-	catcher := grip.NewBasicCatcher()
-	bucket, err := t.GetBucket(ctx)
-	if err != nil {
-		return nil, err
-	}
-
 	switch t.Artifact.Version {
 	case 0:
+		bucket, err := t.GetBucket(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		var results []TestResult
 		iter := NewTestResultsIterator(bucket)
 		for iter.Next(ctx) {
-			results.Results = append(results.Results, iter.Item())
+			results = append(results, iter.Item())
 		}
 
+		catcher := grip.NewBasicCatcher()
 		catcher.Wrap(iter.Err(), "iterating test results")
 		catcher.Wrap(iter.Close(), "closing test results iterator")
+
+		return results, catcher.Resolve()
 	case 1:
-		r, err := bucket.Get(ctx, testResultsCollection)
+		bsonResults, err := t.downloadBSON(ctx)
 		if err != nil {
-			catcher.Wrap(err, "getting test results")
-			break
+			return nil, err
 		}
-
-		data, err := ioutil.ReadAll(r)
-		catcher.Add(r.Close())
-		if err != nil {
-			catcher.Wrap(err, "reading test results")
-			break
+		parquetResults, err := t.downloadParquet(ctx)
+		if err != nil && !pail.IsKeyNotFoundError(err) {
+			return nil, err
+		} else if err == nil && invalidParquetAlertCount < 5 {
+			if reflect.DeepEqual(parquetResults, bsonResults) {
```

Comment:
shouldnt' this be `if ! equal`?

Reply:
Is there any way to see what the (deep) diff actually is?

Reply:
Good catch!

Reply:
I am nervous to log the diff because these test results can be really large for certain projects (sometimes on the order of 100k results).

Reply:
But since we have the taskid/execution, we can retroactively download the test BSON and Parquet results and manually evaluate them.